### ISSUE TITLE
Training Problem

### ISSUE THREAD

**Issue by Transformer-man** [okunator/cellseg_models.pytorch#53]
misconfigurationexception: you passed `accelerator='gpu'`, but you didn't pass `gpus` to `trainer`.

**Comment by okunator** [comment]
Hi,

This is a pytorch-lightning error. Basically, you just need to pass `n_devices` argument to the `pl.Trainer` to fix this.

**Comment by Transformer-man** [comment]
Thank you for your help. But I seem to have encountered another problem. How to import a trained model into prediction。
When I use the following code
_experiment = experiment.load_state_dict(torch.load('./version_0/checkpoints/epoch=149-step=49800.ckpt'))_
![QQ截图20230320112516](https://user-images.githubusercontent.com/56433789/226239499-0a1dc96c-8c07-472f-b2d3-69b57d761273.jpg)

**Comment by okunator** [comment]
Can you be a bit more specific so I can get a hang of what you're trying to achieve. In general, you should not try to load the state dict to the`SegmentationExperiment` object. Instead, you should load the weights into a `nn.Module` object and use that as an input to the `SegmentationExperiment`

**Comment by Transformer-man** [comment]
I have already solved this problem. I would like to ask how to save the best model results on the validation set. In addition, may I ask you how to validate a test set? Thank you very much.

**Comment by okunator** [comment]
You can save your checkpoints with pytorch-lightning `ModelCheckpoint` class. Ory ou can just add a `ckpt_path` to the  `trainenr.fit` method: `trainer.fit(model, ckpt_path="some/path/to/my_checkpoint.ckpt")`. Check out the lightning checkpointing documentation: https://lightning.ai/docs/pytorch/stable/common/checkpointing_intermediate.html

**Comment by okunator** [comment]
For benchmarking: see this benchmarking notebook: https://github.com/okunator/cellseg_models.pytorch/blob/main/examples/pannuke_cellpose_benchmark.ipynb

**Comment by Transformer-man** [comment]
Hello, I'm training pannuke_nuclei_segmentation_stardist.ipynb. I found that you have enhanced the data used for images and masks. The first dimension of the mask is a cell count map, and the second dimension is a n
![QQ截图20230323110244](https://user-images.githubusercontent.com/56433789/227095704-f0123752-f53e-43f4-8758-25d33315006d.jpg)

![QQ截图20230323110221](https://user-images.githubusercontent.com/56433789/227095698-6a1e5fa0-5ff5-4d58-82ec-492e7f5c94d6.jpg)
uclear category map. However, the first dimension is used in your code. Is there a problem?

**Comment by Transformer-man** [comment]
Hello, I'm training [pannuke_nuclei_segmentation_stardist.ipynb](https://github.com/okunator/cellseg_models.pytorch/blob/main/examples/pannuke_nuclei_segmentation_stardist.ipynb), I found that you have enhanced the data used for images and masks. The first dimension of the mask is a cell count map, and the second dimension is a nuclear category map. However, the first dimension is used in your code. Is there a problem?

![QQ截图20230323110221](https://user-images.githubusercontent.com/56433789/227095810-b29cc1b9-96d1-4310-b9d0-bd485fc0648e.jpg)
![QQ截图20230323110244](https://user-images.githubusercontent.com/56433789/227095813-47f37f61-7f87-4d39-9c87-670597a1c6b6.jpg)

**Comment by Transformer-man** [comment]
Should we use a second dimensional cell category mask instead of a cell technology mask?

**Comment by Transformer-man** [comment]
I trained 150 epochs using default parameters, and the test results seemed very unsatisfactory.[pannuke_nuclei_segmentation_stardist.ipynb](https://github.com/okunator/cellseg_models.pytorch/blob/main/examples/pannuke_nuclei_segmentation_stardist.ipynb)
![QQ截图20230323141021](https://user-images.githubusercontent.com/56433789/227118592-70b37182-cd21-4b90-9513-af550158efd1.jpg)

**Comment by okunator** [comment]
> Hello, I'm training [pannuke_nuclei_segmentation_stardist.ipynb](https://github.com/okunator/cellseg_models.pytorch/blob/main/examples/pannuke_nuclei_segmentation_stardist.ipynb), I found that you have enhanced the data used for images and masks. The first dimension of the mask is a cell count map, and the second dimension is a nuclear category map. However, the first dimension is used in your code. Is there a problem?
> 
> ![QQ截图20230323110221](https://user-images.githubusercontent.com/56433789/227095810-b29cc1b9-96d1-4310-b9d0-bd485fc0648e.jpg) ![QQ截图20230323110244](https://user-images.githubusercontent.com/56433789/227095813-47f37f61-7f87-4d39-9c87-670597a1c6b6.jpg)

No there is not a problem, the data flow is such that the instance mask and the semantic masks are transformed separately since they require different kinds of transformations. I'm not quite sure what you mean about enhancing data but applying augmentations during the data loading is a routine practice and if you wish to not use augmentations you don't have to

**Comment by okunator** [comment]
> I trained 150 epochs using default parameters, and the test results seemed very unsatisfactory.[pannuke_nuclei_segmentation_stardist.ipynb](https://github.com/okunator/cellseg_models.pytorch/blob/main/examples/pannuke_nuclei_segmentation_stardist.ipynb) ![QQ截图20230323141021](https://user-images.githubusercontent.com/56433789/227118592-70b37182-cd21-4b90-9513-af550158efd1.jpg)

Stardist is not easy to train on multi-class problems and requires a lot of regularization or other bias mitigation since the learning of the radial distance maps dominate the gradient updates during training. You might want to try out cellpose or hover-net instead for pannuke data.

**Comment by Transformer-man** [comment]
Thank you for your help

**Comment by Transformer-man** [comment]
Hello, but I used the training set to test indicators and found that there are two categories that are also 0. Is there any problem?
![QQ截图20230324203913](https://user-images.githubusercontent.com/56433789/227523791-20839154-ef66-43c7-a9d8-5332a5f8c9c8.jpg)

**Comment by okunator** [comment]
No, it just mean that the model did not learn to recognize those categories.
